pkg/controller/controller.go

```diff
@@ -866,11 +868,46 @@ func convertAndFilterCatalogToNamespacedTypes(namespace string, in *osb.CatalogR
 	return serviceClasses, servicePlans, nil
 }
 
+func generateEscapedName(externalID string) string {
+	startingRE := regexp.MustCompile("^\\.")
+	re := regexp.MustCompile("[^a-z0-9.]")
+	endingRE := regexp.MustCompile("\\.$")
+	invalidHyphens := regexp.MustCompile("--")
+	invalidHyphenPeriod := regexp.MustCompile("-\\.")
+	invalidPeriodHyphen := regexp.MustCompile("\\.-")
```

Comment:
a paragraph explaining all of this in words would be nice. or a bulleted list of the cases that each of these cover.